app/chitter.rb

```diff
@@ -0,0 +1,89 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative '../datamapper_setup.rb'
+require_relative './models/user'
+require_relative './models/peep'
+
+class App < Sinatra::Base
+
+  set :views, proc { File.join(root, '..', 'views') }
+  enable :sessions
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+  use Rack::MethodOverride
+
+  get '/' do
+    erb :index
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :'users/new'
+  end
+
+  post '/users' do
+  	@user = User.create(email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation], name: params[:name], username: params[:username])
+  	if @user.save
+      session[:user_id] = @user.id
+      redirect to('/')
+    else
+      flash.now[:error] = @user.errors.full_messages#'Password does not match'	
```

Comment:
Missing space after #.<br>Put a space before an end-of-line comment.<br>Trailing whitespace detected.